caffe2/serialize/inline_container.h

```diff
@@ -100,6 +101,7 @@ class CAFFE2_API PyTorchStreamReader final {
  public:
   explicit PyTorchStreamReader(const std::string& file_name);
   explicit PyTorchStreamReader(std::istream* in);
+  explicit PyTorchStreamReader(ReaderFunc in, SeekerFunc seeker, size_t size);
```

Comment:
This feels redundant withe ReadAdapterInterface. Both abstract the ability to read. I'd prefer that we only had ReadAdapterInterface. (In fact, I'd prefer that PyTorchStreamReader had only one constructor and the differences between taking a file, a stream, or something else were abstracted in ReadAdapterInterface.

Reply:
`PyTorchStreamReader` just forwards these to construct a `ReadAdapterInterface` (`FuncAdapter`), are you saying this should happen earlier? Doing it as is also allows `PyTorchStreamReader` to be bound directly to Python without any custom weirdness around its constructor

Reply:
PyTorchStreamReader should not need an additional constructor. The implementation of a ReadAdapterInterface that would work with python objects can be put directly in the binding for the constructor of PythonStreamReader.